include/zephyr/fs/fs_interface.h

```diff
@@ -61,6 +61,8 @@ struct fs_mount_t;
  * The object needs to be initialized with fs_file_t_init().
  */
 struct fs_file_t {
+	/** Name of the file */
+	char file_name[MAX_FILE_NAME];
```

Comment:
That is significant increase in size of the fs_file_t structure, is it possible to offload that to POSIX layer, I am not sure that VFS layer is interested in taking the cost, at this point.

Reply:
@de-nordic  But it helps VFS layer too. There is no way to retreive the filename in current vfs implementation. I understand that it increase the fs_file_ struct, but it is from the application layer.

Reply:
Embedding the full filename like this is definitely not ideal.

There is an algorithm to piece together the full path given the inode, dirent info, and mount point.

https://stackoverflow.com/a/33987798

However, judging by [fs_stat()](https://github.com/zephyrproject-rtos/zephyr/blob/45205f865e18decbb9568c4691686dc679cbe961/include/zephyr/fs/fs.h#L606) API and thr output struct, [fs_dirent](https://github.com/zephyrproject-rtos/zephyr/blob/45205f865e18decbb9568c4691686dc679cbe961/include/zephyr/fs/fs.h#L117)

it really doesn't seem like it's sufficiently possible to fill in the majority of [struct stat](https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/sys_stat.h.html) using the `fs_stat()` call unless several additional items were added to the dentry structure (although that really is not where they belong).

I almost wonder if a new fs subsystem call will be required.

Reply:
> @de-nordic But it helps VFS layer too. There is no way to retreive the filename in current vfs implementation. I understand that it increase the fs_file_ struct, but it is from the application layer.

It does not seem there is an issue that requests the addition so far. 
As far as I can tell this is, now, required by POSIX, but there is not final client for that, no user case where addition/subsystem will finally request that kind of info.

The `fs_file_t` is very often allocated on stack or bss, this means that there will be huge addition to stack requirement (or bss)  for barely any use, as the code that does the definition of `fs_file_t` object really knows the file name and it will not even be interested to get it from the file object.
Basically: at the VFS API level users know names of files they tinker with or can store them in their own structures.

We have left all the `fs_file_t` object memory management to users/subsystems, because they know best when and how many objects they need, when the code gets into to the usage case.

POSIX has to pre-define, at compile time, how many (CONFIG_POSIX_MAX_FDS) entries the fd_table will be using and that is something you have tune for entire system; note that this is embedded system so you can not just tell anyone that now they have to get extra ram or do `ps` and find the thing that eats to many fds and kill it for other subsystem to work.

We could probably ifdef this with some Kconfig option, but note that, once you add this, all the `f_file_t` objects will have the array whether anything uses it or not; so if you have one use case for N objects the RAM tax for the usage case is `(N-1) * MAX_FILE_NAME`, for the VFS alone.

I think that rather than this change to `fs_file_t`, there should be RFC for the change explaining what are benefits, and what are costs so that the community could decide on the path forward. Or the `fdtable` should take the cost until VFS use case is found.